WebScraping/docker-lambda-function/lambda_function.py

```diff
@@ -2,13 +2,62 @@
 from tempfile import mkdtemp
 from selenium.webdriver.support.ui import Select
 from selenium.webdriver.common.by import By
+from time import sleep
 from bs4 import BeautifulSoup
+from unicodedata import normalize
+import json
+import boto3
 import re
 
 
+PAGE_SOURCE = "https://central.carleton.ca/prod/bwysched.p_select_term?wsea_code=EXT"
+CHROME_SERVICE_EXE_PATH = "/opt/chromedriver"
+SEARCH_XPATH = r'//*[@id="ws_div"]/input'
+CHANGE_TERM_BTN_XPATH = "/html/body/section/section/form/table/tbody/tr[5]/td/input[4]"
+SEARCH_BTN_XPATH = "/html/body/section/section/form/table/tbody/tr[5]/td/input[1]"
+RETURN_TO_SEARCH_XPATH = "/html/body/section/section/form/table/tbody/tr[5]/td/input[2]"
+HREF_STR = "bwysched.p_display_course?"
+LINKED_COURSE_STR = "https://carleton.ca/registrar/registration/terminology/"
+BUCKET_NAME = "carletonschedulingtool"
+KEY_PATH = "web-scraping-stepfunction/href_list.json"
+
 def handler(event=None, context=None):
+    driver = init()
+
+    class_info = []
+
+    terms = scrape_terms(driver)
+    
+    for term in terms:
+
+        select_terms(driver, term)
+
+        options = scrape_course_options(driver)
+
+        for option in options:
+
+            if option == "":
+                continue
+
+            select_options(driver, option)
+
+            # Grab course href and linked courses
+            class_info = parse(html_page=driver.page_source, lst=class_info)
```

Comment:
Here you are passing the list "class_info" as a parameter to the parse function. Passing a list is "pass-by-reference" meaning that any changes in the list class_info will be reflected without assigning class_info = parse(html_page=driver.page_source, lst=class_info). I suggest either changing this line to just "parse(driver.page_source, class_info)" or do not pass the list as a parameter and do "class_info.extend(parse(driver.page_source))". Also, is there a reason you are passing the arguments using "=", usually this only done if there are keyword arguments?

Reply:
The following line will be changed to "parse(driver.page_source, class_info)", and there is no reason for adding the "=" when passing in the argument, so that will be removed as well.